internal/authz/oidc.go

```diff
@@ -396,6 +396,81 @@ func (o *oidcHandler) retrieveTokens(ctx context.Context, log telemetry.Logger,
 	setDenyResponse(resp, deny, codes.Unauthenticated)
 }
 
+// buildAuthHeader builds the authorization header for the client according to the
+// client authentication method specified in the OIDCConfig.
+//
+// The function returns an error if the client authentication method is unspecified
+// or if the implementation for the specified method is not supported.
+func buildAuthHeader(config *oidcv1.OIDCConfig) (http.Header, error) {
+
+	headers := http.Header{}
+	switch config.GetMethod() {
+	case oidcv1.OIDCConfig_CLIENT_AUTHENTICATION_METHOD_BASIC:
```

Comment:
Take care. In Go there is no automatic fallthrough! the `default` block will _NOT_ be executed if the client config method is set to basic.